test/libp2p_test.exs

```diff
@@ -71,4 +89,73 @@ defmodule Libp2pTest do
     :ok = Libp2p.host_close(sender)
     :ok = Libp2p.host_close(recver)
   end
+
+  test "discover peer and add it to peerstore" do
+    {:ok, host} = Libp2p.host_new()
+
+    {:ok, peerstore} = Libp2p.host_peerstore(host)
+
+    {:ok, listener} =
+      Libp2p.listen_v5("0.0.0.0:25000", @bootnodes)
+
+    {:ok, iterator} = Libp2p.listener_random_nodes(listener)
+
+    true = Libp2p.iterator_next(iterator)
+    {:ok, node} = Libp2p.iterator_node(iterator)
+
+    {:ok, id} = Libp2p.node_id(node)
+    {:ok, addrs} = Libp2p.node_multiaddr(node)
+
+    :ok = Libp2p.peerstore_add_addrs(peerstore, id, addrs, Libp2p.ttl_permanent_addr())
+
+    :ok = Libp2p.host_close(host)
+  end
```

Comment:
Not for this PR, as what you did is perfectly fine for a think-wrapper. We should add a different layer of wrapping were a `listener_random_nodes` returns an [Stream](https://hexdocs.pm/elixir/1.12/Stream.html) elixir type that we can map/flatmap, etc over.

Reply:
Thanks for the comment! I didn't know about [`Stream.resource`](https://hexdocs.pm/elixir/1.12/Stream.html#resource/3) 😮. It makes it really easy to implement the wrapper!